src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,102 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const handleAfterSubmitting = () => {
```

Comment:
Using more descriptive names like `resetFormAfterSubmit `makes the code easier to understand by clearly indicating the function's purpose. It improves maintainability, helping others quickly grasp what the function does without needing to read its implementation.
```suggestion
  const resetFormAfterSubmit = () => {
```